File: kora-app-annotation-processor/src/test/java/ru/tinkoff/kora/kora/app/annotation/processor/app/AppWithOverridenModule.java
package ru.tinkoff.kora.kora.app.annotation.processor.app;

import ru.tinkoff.kora.common.KoraApp;
import ru.tinkoff.kora.common.annotation.Root;

@KoraApp
public interface AppWithOverridenModule extends Module2 {
    @Root
    default Class1 class1(Class2 class2) {
        return new Class1(class2);
    }

    record Class1(Class2 class2) {}

    record Class2() {}


}

interface Module {
    default AppWithOverridenModule.Class2 class2() {
        throw new IllegalStateException();
    }
}


interface Module2 extends Module {
    @Override
    default AppWithOverridenModule.Class2 class2() {
        return new AppWithOverridenModule.Class2();
    }
}

File: kora-app-symbol-processor/src/test/kotlin/ru/tinkoff/kora/kora/app/ksp/app/AppWithOverridenModule.kt
package ru.tinkoff.kora.kora.app.ksp.app

import ru.tinkoff.kora.common.KoraApp
import ru.tinkoff.kora.common.annotation.Root

@KoraApp
interface AppWithOverridenModule : Module2 {
    @Root
    fun class1(class2: Class2): Class1 {
        return Class1(class2)
    }

    data class Class1(val class2: Class2)

    class Class2
}

interface Module {
    fun class2(): AppWithOverridenModule.Class2 {
        throw IllegalStateException()
    }
}

interface Module2 : Module {
    override fun class2(): AppWithOverridenModule.Class2 {
        return AppWithOverridenModule.Class2()
    }
}
